### PR TITLE
[DNM] store/tikv: remove cluster ID

### PR DESCRIPTION
**Pull request by huachaohuang**
Ref: https://github.com/pingcap/pd/pull/372

Need to update tidb-binlog after this has been merged.

### CODE REVIEW

**Comment by huachaohuang** [comment]
Update vendor after https://github.com/pingcap/pd/pull/373.

**Comment by huachaohuang** [comment]
PTAL @siddontang @coocood

File: store/tikv/kv.go [hunk 1]
```diff
@@ -84,13 +87,14 @@ type tikvStore struct {
 	gcWorker     *GCWorker
 }
 
-func newTikvStore(uuid string, pdClient pd.Client, client Client, enableGC bool) (*tikvStore, error) {
+func newTikvStore(clusterID uint64, uuid string, pdClient pd.Client, client Client, enableGC bool) (*tikvStore, error) {
```
Comment (coocood) [inline store/tikv/kv.go]: This parameter seems not needed. We can get clusterID from pdClient.

File: store/tikv/mock-tikv/pd.go [hunk 2]
```diff
@@ -41,6 +41,10 @@ func NewPDClient(cluster *Cluster) pd.Client {
 	}
 }
 
+func (c *pdClient) GetClusterID() uint64 {
```
Comment (coocood) [inline store/tikv/mock-tikv/pd.go]: `GetClusterID` may returns error?

**Comment by huachaohuang** [comment]
PTAL @coocood @disksing

File: store/tikv/kv.go [hunk 3]
```diff
@@ -50,20 +49,22 @@ func (d Driver) Open(path string) (kv.Storage, error) {
...
 	}
+
+	pdCli, err := pd.NewClient(etcdAddrs)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
 	// FIXME: uuid will be a very long and ugly string, simplify it.
-	uuid := fmt.Sprintf("tikv-%v-%v", etcdAddrs, clusterID)
+	uuid := fmt.Sprintf("tikv-%v-%v", etcdAddrs, pdCli.GetClusterID())
```
Comment (disksing) [inline store/tikv/kv.go]: Seems better to use only the clusterID, since 'host1:port1,host2:port2' and 'host2:port2,host1:port1' refer to a same cluster.

**Comment by coocood** [comment]
LGTM

**Pull request merged**